builtin/providers/nsone/provider.go

```diff
@@ -32,9 +35,10 @@ func Provider() terraform.ResourceProvider {
 }
 
 func nsoneConfigure(d *schema.ResourceData) (interface{}, error) {
-	n := nsone.New(d.Get("apikey").(string))
-	n.Debug()
-	n.RateLimitStrategySleep()
+	httpClient := &http.Client{}
+	n := nsone.NewClient(httpClient, nsone.SetAPIKey(d.Get("apikey").(string)))
+	// FIXME: n.Debug()
```

Comment:
Sorry, trying to find docs on terraform debug state now. We can adding whatever logs to the requests/responses that we want with the decorator. my guess is that it should be logging both requests and responses. is their any documentation on the debug state?